frontend/src/components/GameReports/rows.tsx

```diff
@@ -0,0 +1,287 @@
+                {
+                    key: "competition-type",
+                    content: summarizeCompetitionType(
+                        report.match,
+                        report.competition
+                    ),
+                },
+                {
+                    key: "league",
+                    content: report.league ? getLeagueLabel(report.league) : "",
+                },
+                {
+                    key: "expansions",
+                    content: report.expansions
+                        .map(getExpansionLabel)
+                        .join(", "),
+                },
+                { key: "tokens", content: report.actionTokens },
+                { key: "dwarven-rings", content: report.dwarvenRings },
+                { key: "corruption", content: report.corruption },
+                { key: "mordor", content: report.mordor },
+                { key: "aragorn", content: report.aragornTurn },
+                { key: "treebeard", content: report.treebeard && "✓" },
+                { key: "eyes", content: report.initialEyes },
+                {
+                    key: "sp-settlements",
+                    content: (
+                        <ShadowCaptures
+                            report={report}
+                            layout={settings.settlementLayout}
+                            isAbbreviated={settings.areSettlementsAbbreviated}
+                        />
+                    ),
+                },
+                {
+                    key: "spvp",
+                    content: countVictoryPoints(
+                        report.strongholds,
+                        report.expansions,
+                        "Free"
+                    ),
+                },
+                {
+                    key: "fp-settlements",
+                    content: <FreeCaptures report={report} />,
+                },
+                {
+                    key: "fpvp",
+                    content: countVictoryPoints(
+                        report.strongholds,
+                        report.expansions,
+                        "Shadow"
+                    ),
+                },
+                { key: "interest", content: report.interestRating },
+                {
+                    key: "comments",
+                    content: report.comment ? (
+                        <CommentText>{report.comment}</CommentText>
+                    ) : null,
+                },
+                {
+                    key: "log",
+                    content: report.logFile && (
+                        <ExternalLink isDownload href={report.logFile}>
+                            Download Report
+                        </ExternalLink>
+                    ),
+                },
+            ].filter(isDefined),
+        })
+    );
+}
+
+interface FixedWidthTextProps {
+    children: ReactNode;
+    width: number;
+}
+
+function FixedWidthText({ width, children }: FixedWidthTextProps) {
```

Comment:
I want to split the 3 sub-components here into their own files too, but trying to be marginally considerate about the size of the diff! Unless you object to more component files, I'll tack on a commit to split them off after you review/before I merge it (don't feel like that part needs a second set of eyes).